File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/SelectResultRequestHandler.swift
//
//  SelectResultRequestHandler.swift
//  CBLTestServer-iOS
//
//  Created by Raghu Sarangapani on 1/16/18.
//  Copyright © 2018 Raghu Sarangapani. All rights reserved.
//

import Foundation
import CouchbaseLiteSwift

public class SelectResultRequestHandler {
    public static let VOID: String? = nil
    fileprivate var _pushPullReplListener:NSObjectProtocol?
    
    public func handleRequest(method: String, args: Args) throws -> Any? {
        switch method {
        //////////////////
        // selectResult //
        //////////////////
        case "selectResult_expressionCreate":
            let expression: Expression = args.get(name: "expression")!
            
            return SelectResult.expression(expression)
            
        case "selectResult_all":
            return SelectResult.all()

        default:
            throw RequestHandlerError.MethodNotFound(method)
        }
        return SelectResultRequestHandler.VOID
    }
}

File: CBLClient/Apps/CBLTestServer-Android/app/src/main/java/com/couchbase/CouchbaseLiteServ/server/RequestHandler/SelectResultRequestHandler.java
package com.couchbase.CouchbaseLiteServ.server.RequestHandler;


import com.couchbase.CouchbaseLiteServ.server.Args;
import com.couchbase.lite.Expression;
import com.couchbase.lite.SelectResult;

public class SelectResultRequestHandler {

    public SelectResult expressionCreate(Args args){
        Expression expression = args.get("expression");
        return SelectResult.expression(expression);
    }

    public SelectResult all(Args args){
        return SelectResult.all();
    }
}
